### ISSUE TITLE
Validate Jira configuration on startup

### ISSUE THREAD

**Issue by xnutsive** [gravitational/teleport-plugins#59]
When you start `teleport-jirabot`, it'll start and log that it's listening for new requests even if the jira part of the config has invalid values, specifically the Jira URL, the project key, the custom access request id field, and the API token.

Then, when a new request pops up, the bot would fail with an error message. 

**Suggestion**: since we can check if most of the config values are setup correctly in advance, we should check them and fail on startup, instead of waiting for a request. specifically: 

- [ ] Check if the project with the provided key exists and is accessible with the provided API token.
- [ ] Check if the custom field (`TeleportAccessRequestId`) exists and is included in the issues of type `Task` in the provided project.
- [ ] Check that the provided Jira URL is a valid Jira Cloud or a Jira Server url.

**Comment by marshall-lee** [comment]
@xnutsive 

Hi! See commit messages bb10473 and 640b73d for more details.

tl;dr:
- Custom field check is skipped since current implementation does not use custom field at all.
- JIRA server URL is impossible to validate correctly since base URL is in fact configurable in JIRA settings. It's incorrect to just check for `/jira` prefix. So I implemented a bunch of "api health checks" that are executed on bot start up. Health checks try to do some suggestions about misconfigured parameters, you can try it out!
